Consistent navigation

manuelmeister opened the pull request:
Die Desktop- und Mobile-Navigation wurden optimiert. Auf Mobile ist das zentrale Element die Suchbar auf der Camp Seite ( neu die Picasso/EventList Seite)
Mit der CardView einen generischen Komponent für den Hauptinhalt erstellt.
Versuch den Loading-Filter zu erstellen.
 
Fixes #360, fixes #263, fixes #361, fixes #366

manuelmeister committed:
Updated inter-ui with the reduced charset

manuelmeister committed:
Moved definitions into different file (otherwise variables.scss are always included and the definitions are duplicated).

manuelmeister committed:
Remove background skeleton loaders.

manuelmeister committed:
Shows already loaded data and better styling of the day headers

manuelmeister committed:
Created CardView as a container for the main content

manuelmeister committed:
Created CardView as a container for the main content

manuelmeister committed:
Improved text on home & implemented new CardView

manuelmeister committed:
Added searchbar (only mockup)

manuelmeister committed:
Fixed styling

manuelmeister committed:
Added CardView and readonly api input

manuelmeister committed:
Implemented CardView

manuelmeister committed:
Added username similar as to isLoggedin

manuelmeister committed:
Added loading filter (probably not yet correctly implemented, but works for now)

manuelmeister committed:
Improved frontend-routing

manuelmeister committed:
Implemented CardView

manuelmeister committed:
Changed import style

manuelmeister committed:
Implemented CardView

manuelmeister committed:
Changed Periods from Component to View

manuelmeister committed:
Added profile with email (read-only)
TODO: Implement backend-methods to change e-mail address

manuelmeister committed:
Changed background color to standard material color

manuelmeister committed:
Implemented CardView and optimized for mobile with toggle between picasso (calendar-view) and listview

manuelmeister committed:
Created navigation that is more consistent

manuelmeister committed:
Updated dependencies

manuelmeister committed:
Added profile link in login

manuelmeister committed:
Linted

manuelmeister committed:
Fixed conflicts from devel
Decided to immediately show loaded instances in picasso.

carlobeltrame commented on frontend/src/components/CardView.vue:
```diff
@@ -0,0 +1,31 @@
+<template>
```
@usu will in https://github.com/pmattmann/ecamp3/pull/4/files#diff-2a3a648439e08b249166925d614536d5 eine sehr ähnliche Komponente hinzufügen. Seine Version ermöglicht zusätzlich noch ein Icon in der Title Bar, deine prüft dafür schöner ob der title gesetzt ist und hat einen eingebauten skeleton loader. Können wir diese beiden unterschiedlich benannten Entwürfe noch vereinen?

manuelmeister replied:
Kein Problem. Ich glaube ich hatte einfach gemerkt, dass wir nicht generell immer eine Toolbar wollen und somit diese wieder entfernt. (die hatte auch ein Icon)
Ich hatte CardView als Namen verwendet, weil ich gedacht habe, dass wir nicht verschiedene Karten auf dem Hintergrund haben sonder eher Gruppen innerhalb der Karte.

carlobeltrame commented on frontend/src/layouts/DefaultLayout.vue:
```diff
@@ -1,64 +1,54 @@
 <template>
```
Die Login-Seiten verwenden noch das EmptyLayout. Dort ist die Title Bar noch manuell in den Views (Login.vue, Register.vue, RegisterDone.vue) eincodiert. Dadurch springt das Logo noch, wenn man ausgeloggt ist und die Seite neu lädt.
Ich würde vorschlagen, das EmptyLayout in ein NotLoggedInLayout umzufunktionieren, und die Navigations-Leiste von den Views dort hinein zu verschieben. Und möglicherweise das eCamp-Logo (z.B. mit boolean Prop "showText") gleich in eine Komponente extrahieren, damit das in Zukunft auf allen Views konsistent aussieht.

manuelmeister replied:
Ich hatte mir noch gedacht, die haben noch nicht das neue Layout angewendet, habe sie aber auch nicht geprüft. Merci für den Hinweis.
Ja gibt Sinn! Super Idee.

manuelmeister replied:
@carlobeltrame Mich stört, dass wenn ich zB. die Periods neu lade, dass zuerst das DefaultLayout geladen wird, kurz der Screen weiss wird und dann erst das CampLayout angezeigt wird.

carlobeltrame replied:
Du könntest das Problem mal lindern indem du die `<v-app>` Tags aus den Layouts ins App.vue verschiebst. Dann flasht der Screen nicht mehr weiss sondern in der Hintergrundfarbe.

Aber die wirklich richtige Lösung wäre wohl, doch nur ein Layout zu haben welches oben die schwarze Navigations-Bar hat, und dann [named router views](https://router.vuejs.org/guide/essentials/named-views.html) für Elemente welche sich unterscheiden können (z.B. `navigation` für Camp-Titel, Team, Admin etc., `user` für das Profil-Dropdown und `default` für den Page Body)

carlobeltrame commented on frontend/src/router.js:
```diff
@@ -64,20 +71,20 @@ export default new Router({
-          meta: { layout: 'camp' }
+          path: '',
+          name: 'camp',
+          redirect: { name: 'camp/picasso' }
```
Cooles Konzept mit dem redirect, aber so funktioniert es leider nicht mehr wenn man /camps/3ddfdcd/admin eingibt. Ich habs nur ohne redirect hingebracht, und nur wenn die Route mit `path: ''` zuletzt im `children` array steht.
Ich finde es persönlich auch gar nicht schlimm wenn auf der normalen Camp-Ansicht hinten nicht noch /picasso steht. Der User klickt in der aktuellen Navigation ja auch nicht auf einen Text "Picasso", sondern auf den Camp-Namen, und der Picasso ist einfach die Default-Ansicht eines Camps.

manuelmeister replied:
Ja stimmt, ich hatte nur gedacht, es ist möglich, dass Benutzer einfach auf das Camp verweisen und /picasso irgendwie verlohren geht. Aber vielleicht kann man das ja anders lösen.

carlobeltrame commented on frontend/src/router.js:
```diff
@@ -50,11 +56,12 @@ export default new Router({
       components: {
         default: () => import(/* webpackChunkName: "camp" */ './views/Camp.vue'),
-        aside: () => import(/* webpackChunkName: "camps" */ './views/Camps.vue')
+        aside: () => import(/* webpackChunkName: "periods" */ './views/Periods.vue')
```
Sehr gute Idee, die Camp-Liste war hier unnötig, und so kann man die Perioden-Tabs vom Picasso-Bereich entfernen :+1: 
Aber: Wie kann ich auf Mobile zwischen Perioden umschalten?

manuelmeister replied:
Ich hatte gedacht durch Swipen oder via das Layer Icon.

carlobeltrame replied:
Passt, können wir dann in Zukunft ja mal anschauen.

carlobeltrame commented on frontend/src/components/camp/CampPicasso.vue:
```diff
@@ -3,47 +3,96 @@ Listing all event instances of a single period.
+          <v-icon v-if="picassoFormat">mdi-menu</v-icon>
+          <v-icon v-else>mdi-calendar-month</v-icon>
+        </v-btn>
+        <picasso
```
Den Skeleton Loader für den Picasso gibts jetzt nicht mehr. War das bewusst? Mir persönlich ist es ehrlich gesagt so fast lieber, mit den Blöcken die mit loading... angeschrieben sind. Ich könnte mir höchstens einen skeleton loader für mehrere zufällig verteilte calendar events vorstellen (noch bevor wir wissen wo im Picasso EventInstances sein werden), aber habe auf die Schnelle in Vuetify keinen solchen gefunden.

manuelmeister replied:
Ja, ich habe gemerkt, dass man erwartet, dass es bereits wie ein Kalender aussieht, wenn es lädt. Somit habe ich ihn entfernt und darauf geachtet, dass ein Wert angezeigt wird, bevor der Event geladen ist.

carlobeltrame commented on frontend/src/main.js:
```diff
@@ -8,6 +8,13 @@ import vuetify from './plugins/vuetify'
 Vue.component('default-layout', () => import('./layouts/DefaultLayout'))
 Vue.component('camp-layout', () => import('./layouts/CampLayout'))
+Vue.filter('loading', function (value, loadingState) {
+  if (typeof value === 'function' && value().loading) {
```
**tl;dr:** Um einen Konsolen-Fehler zu vermeiden, schreibe in storeValueProxy.js auf Zeile 80 (unter `result.toString = () => ''`) folgende Zeile dazu: `result.loading = true`, und ändere die Bedingung hier in `if (typeof value === 'function' && value.loading)` (oder übernimm gleich die Version von mir im Kommentar in Picasso.vue).

**Erklärung wieso das nötig ist:** Meistens in diesem Filter wird `value` eine primitive Property einer Entity sein. Solange `value` noch aus einem loadingProxy gezogen wird ist diese if-Bedingung passend: `value()` retourniert einen loadingProxy, und dieser reagiert auf `.loading` mit `true`. Allerdings triggert der Funktionsaufruf `value()` intern im loadingProxy noch ein `.then(...)`, welches später ausgeführt wird sobald der echte Wert von `value` aus der API geladen ist (z.B. der String 'Camp 1 Title'). Dieses `.then(...)` versucht dann noch einmal deine Aktionen mit `value` nachzustellen, also auch den Funktionsaufruf `value()`. Das gibt aber einen Konsolenfehler, weil `'Camp 1 Title'()` nicht funktioniert.
Ich sehe mehrere Lösungen dafür, aber als schnelle und halbwegs saubere Lösung würde ich glaube ich direkt auf `value` markieren dass da noch etwas am laden ist. Dazu die Anweisungen aus dem tl;dr ausführen.

carlobeltrame commented on frontend/src/components/camp/CampPicasso.vue:
```diff
@@ -3,47 +3,96 @@ Listing all event instances of a single period.
+          <v-avatar size="32" class="ml-4">
+            <img
+              alt="Avatar"
+              src="https://avatars0.githubusercontent.com/u/9064066?v=4&s=460">
```
Wer ist das? :rofl:

manuelmeister replied:
Das ist der Gründer von Vuetify. War ein Beispiel auf der Seite und ich wollte verdeutlichen, dass dort ein Avatar ist.

carlobeltrame commented on frontend/src/components/camp/CampPicasso.vue:
```diff
@@ -3,47 +3,96 @@ Listing all event instances of a single period.
   data () {
     return {
+      searchOpen: false,
+      picassoFormat: true,
```
Diese Einstellung geht verloren, wenn man in der Listenansicht (`picassoFormat === false`) einen Block anklickt und dann zurücknavigiert. Ich würde vorschlagen picassoFormat in ein computed umzuwandeln welches `this.$route.query.list` oder so liest, und beim Toggle-Button das `@click` durch `:to="{ query: { ...$route.query, list: !picassoFormat } }">` zu ersetzen.

manuelmeister replied:
Versuche das Mal :)

carlobeltrame commented on frontend/src/layouts/CampLayout.vue:
```diff
@@ -152,6 +156,9 @@ export default {
       return campFromRoute(this.$route)
+    },
+    username () {
+      return this.$auth.username()
```
Könntest du hier und im DefaultLayout stattdessen `this.api.get().profile().username` verwenden? Dann müsstest du nicht in auth.js und dem Backend den username zum Auth endpoint hinzufügen. Falls der User nicht eingeloggt ist würde ich das sowieso mit einem anderen Layout abhandeln, welches gar nicht erst versucht, den username anzuzeigen.

carlobeltrame commented on frontend/src/scss/global.scss:
```diff
@@ -0,0 +1,29 @@
+.v-skeleton-loader__table-thead,
+.v-skeleton-loader__table-tbody,
+.v-skeleton-loader__table-tfoot {
+	background: none!important;
```
Ich sehe dass das in der Sidebar der Block-Detailansicht einen Unterschied macht. Haben wir vielleicht dort die falsche Art von skeleton loaders eingesetzt? Ich würde solche Modifikationen nur im Notfall vornehmen.

manuelmeister replied:
Nein, ich finde es irgendwie seltsam, dass sie im SkeletonLoader den Hintergrund auf weis setzten. Das gleiche Problem tritt auch auf, wenn man eine Toolbar mit Elevation hat. So wird der Boxshadow nicht angezeigt weil der SkeletonLoader mit seinem durchgehend-weissen Hintergrund den Shadow überdeckt.

carlobeltrame replied:
Alles klar, passt so für mich in dem Fall. Wenigstens schön zentralisiert abgehandelt.

carlobeltrame commented on frontend/src/views/Home.vue:
```diff
@@ -1,17 +1,20 @@
+  <card-view title="Herzlich willkommen bei eCamp" max-width="900">
     <v-list class="pt-0">
-      <v-list-item :to="{ name: 'camps' }">
+      <v-list-item v-if="loggedIn" :to="{ name: 'camps' }">
```
Nicht nötig oder? Wenn man nicht eingeloggt ist wird man sowieso direkt auf die Login-Page umgeleitet.

carlobeltrame commented on frontend/src/views/Home.vue:
```diff
@@ -1,17 +1,20 @@
+          <v-list-item-title>Meine Camps</v-list-item-title>
+        </v-list-item-content>
+      </v-list-item>
+      <v-list-item v-if="!loggedIn" :to="{ name: 'login' }">
```
Das hier auch entfernen, das v-if wird nie erfüllt sein

carlobeltrame commented on frontend/src/views/Home.vue:
```diff
@@ -41,13 +44,19 @@
   name: 'Home',
+  components: { CardView },
   computed: {
+    loggedIn () {
```
Ebenfalls entfernen

carlobeltrame commented on frontend/src/views/Periods.vue:
```diff
@@ -0,0 +1,51 @@
+      Periods
+    </v-subheader>
+    <v-list>
+      <v-list-item
```
Würde es Sinn machen das klickbar zu machen, sodass diese Periode im Picasso geöffnet wird?

manuelmeister replied:
Ja, habe mir aber die Edgecases noch nicht überlegt. zB 2 Perioden à 2 Tage anstatt eine Periode à 2 Tage und eine à 7 Tage. Bei 2x2 würd evtl sinn machen, alle Tage einzeln nebeneinander anzuzeigen.
Ich glaube das ist noch viel Magic to be done.

manuelmeister replied:
Neues Issue erfasst #386

carlobeltrame commented on frontend/src/views/Periods.vue:
```diff
@@ -0,0 +1,51 @@
+  methods: {
+    eventInstanceRoute (eventInstance) {
+      return eventInstanceRoute(this.camp(), eventInstance)
+    }
```
Entfernen, braucht es alles nicht auf dieser Ansicht

carlobeltrame commented on frontend/src/components/camp/CampPicasso.vue:
```diff
@@ -3,47 +3,96 @@ Listing all event instances of a single period.
+          <v-icon v-if="picassoFormat">mdi-menu</v-icon>
+          <v-icon v-else>mdi-calendar-month</v-icon>
+        </v-btn>
+        <picasso
```
Noch ein anderes Problem wenn du den skeleton loader für den ganzen Picasso nicht mehr hast: Ich könnte mir vorstellen dass beim produktiven Build das computed `events` rausoptimiert wird (habs aber nicht ausprobiert). Falls das der Fall wäre (oder jemand in Zukunft das nun nicht mehr verwendete computed "aufräumt"), dann werden wieder alle Events in einzelnen Netwerk-Requests geladen. Vorhin hatte ich das ja verhindert indem ich den Skeleton Loader mit `v-if="events._meta.loading"` markiert hatte.

manuelmeister replied:
Hm, evtl eine Komponente als Loader kreieren?

carlobeltrame commented on frontend/src/components/EventList.vue:
```diff
@@ -0,0 +1,50 @@
+    return {
+    }
+  },
+  computed: {
```
Hier brauchts wohl auch noch ein `mounted` welches die `events()` mit einem einzigen Netzwerk-Request lädt.

manuelmeister replied:
Nein, weil es nur eine Komponente und kein View ist (REF: https://github.com/ecamp/ecamp3/pull/376#discussion_r387652963)

carlobeltrame commented on frontend/src/components/Picasso.vue:
```diff
@@ -48,11 +50,20 @@ export default {
     }
   },
+  mounted () {
+    this.camp().events()
```
Macht das wirklich Sinn das nach hier zu verschieben? Diese Komponente wird ja auch in der Sidebar in der Detailansicht eines Blocks angezeigt. In dem Fall müsste man aus meiner Sicht nicht alle Events laden, sondern nur die die im Tag vorkommen (also die die in der Prop eventInstances sowieso schon von aussen reinkommen). Verschiebung in `mounted` finde ich okay, aber sollte in CampPicasso.vue sein anstatt hier.

manuelmeister replied:
Stimmt 👍

usu replied:
Yep, sehe ich auch so. Ausserdem könnte man dann `CampPicasso.vue` gleich nach `views` verschieben anstatt in `components`. Dann hat man eine klare Trennung. `CampPicasso` ist zuständig für Laden der Daten und generellen Seitenaufbau. Die Komponente `Picasso` schaut lediglich dafür, dass die von aussen übergebenen eventInstances richtig gerendert werden.

carlobeltrame commented on frontend/src/components/Picasso.vue:
```diff
@@ -3,28 +3,30 @@ Listing all given event instances in a calendar view.
+    event-start="start_time"
+    event-end="end_time"
+    :event-name="getEventName"
+    :event-color="getEventColor"
```
Theoretisch kann man in Vue auch hier in Props den `loading` Filter benutzen. Hier ist das leider nicht ganz so einfach, weil wir eine Funktion in die Prop reingeben. Würden wir den bestehenden Filter benutzen, so bekäme er nur die Funktion `getEventName` als Argument, nicht das Resultat der Funktion. Man könnte aber den Filter so erweitern, dass er merkt wenn er eine Funktion als Input bekommt und diese wrappt. Ich habe das mal so versucht:
```js
Vue.filter('loading', function (value, loadingState, isLoading = v => typeof v === 'function' && v.loading) {
  if (typeof value === 'function' && !value.loading) {
    // Wrap the function that is passed into the | loading filter
    return (v, ...args) => isLoading(v) ? loadingState : value(v, ...args)
  }
  return isLoading(value) ? loadingState : value
})
```

Dann könnten wir die loading Logik aus `getEventName` und `getEventColor` rausnehmen. Die passt in meinen Augen nicht wirklich dort hin.

```suggestion
    :event-name="getEventName | loading('Lädt…', ({ input }) => input.event()._meta.loading)"
    :event-color="getEventColor | loading('grey lighten-2', ei => ei.event()._meta.loading)"
```

manuelmeister replied:
Ja wenn das so gut funktioniert, dann kann ich das machen. Aber @carlobeltrame ich verstehe den Code noch nicht.

carlobeltrame replied:
Ja ist leider nicht mehr ganz simpel. Ich probiere es zu erklären.
* Ich würde am liebsten schreiben: `:event-name="getEventName | loading('Lädt...')"`.
* Vuetify erwartet in der `event-name` Prop eine Funktion, also muss `getEventName | loading('Lädt...')` eine Funktion als Resultat haben.
* Wenn man `getEventName | loading('Lädt...')` schreibt, dann ruft Vue unseren Filter mit der Funktion `getEventName` als erstes Argument (`value`) und dem String `'Lädt...'` als zweites Argument (`loadingState`) auf.
* Jetzt müssen wir als erstes mal erkennen, ob da jemand einen normalen Wert (`camp.name | loading(...)`) oder eine Funktion (`getEventName | loading(...)`) in den `loading` Filter füttert. Wenn der Typ des ersten Arguments `function` ist, dann könnte es immer noch sein, dass das nur ein vorläufiges Resultat eines loadingProxies ist. Darum schauen wir zusätzlich noch, ob das `loading` Flag abwesend ist, welches vom loadingProxy gesetzt werden würde.
* Im `if` drin wissen wir also, dass eine echte, bekannte Funktion in den `loading` Filter gegeben wurde. Also müssen wir jetzt auch eine Funktion als Resultat ausspucken.
* Unsere resultierende Funktion muss ein oder mehrere Argumente akzeptieren, genau so viele wie `getEventName` nehmen würde. Wir wissen aber nicht genau wie viele das sind, daher verwenden wir `...args` um einfach alle verfügbaren Argumente in einem Array zu bekommen.
* Uns interessiert später das erste Argument besonders, darum benenne ich es separat `v`.
* Nun muss ich in der neuen Funktion drin entscheiden, ob die Argumente (z.B. die EventInstance die Vuetify da reinfüttern will) noch am laden sind oder nicht. Das Problem ist: Die Definition von "geladen" kann von Benutzung zu Benutzung des Filters unterschiedlich sein. Bei `getEventName` gilt der Input als geladen, sobald `v.input.event()` geladen ist. Bei `getEventColor` hingegen sobald `v.event()` geladen ist. Da ich das nicht im `loading` Filter drin wissen kann, biete ich die Möglichkeit, von aussen (via `isLoading`) dieses Verhalten zu steuern.
* Falls `isLoading` befindet, dass das erste Argument (`v`) noch am laden ist, dann gebe ich den `loadingState` als Resultat an Vuetify zurück.
* Falls `isLoading` hingegen befindet, dass das erste Argument fertig geladen ist, kann ich es durch die eigentliche Funktion (`getEventName`, welche ja in `value` steht) füttern und das tatsächliche finale Resultat an Vuetify zurückgeben.

Weiss jetzt nicht ob dir die Textwand viel hilft. Vielleicht fällt mir auch noch ein Weg ein, den Code etwas aufzusplitten und so verständlicher zu machen.

carlobeltrame commented on frontend/src/components/camp/CampPicasso.vue:
```diff
@@ -3,47 +3,96 @@ Listing all event instances of a single period.
-    </v-tabs-items>
-  </v-card>
+  <card-view>
+    <div v-for="period in periods.items"
```
Brauchts dieses `div` mit `v-for`? So wie ich das verstehe erstellt das eine separate `v-app-bar` für jede Periode...

manuelmeister replied:
Ja, das war nur, damit alles angezeigt wird (in der Entwicklung)

usu replied:
Also die ganze `v-app-bar` mit search und profile wird hier ja sowieso nicht angezeigt, oder? Der Code kommt dann wieder weg?

manuelmeister replied:
@usu nein, die wird auf Mobile nur in diesem View angezeigt. Ich glaube das gibt nicht Sinn, die die TopBar Komponente der Camps zu packen.

carlobeltrame commented on frontend/src/layouts/CampLayout.vue:
```diff
@@ -1,95 +1,97 @@
+          <v-toolbar-items>
+            <v-btn right text
+                   :class="{ 'v-btn--open': value }" v-on="on">
+              <span v-if="loggedIn" class="sr-only-sm-and-down">
```
Auch hier würde ich `v-if="loggedIn"` und das dazugehörige computed entfernen, diese Ansicht ist mit einem Camp verbunden und darauf sollte man gar nie kommen wenn man nicht eingeloggt ist.

manuelmeister replied:
Ah ja stimmt :)

carlobeltrame commented on frontend/src/layouts/DefaultLayout.vue:
```diff
@@ -1,64 +1,54 @@
+          <v-toolbar-items>
+            <v-btn right text
+                   :class="{ 'v-btn--open': value }" v-on="on">
+              <span v-if="loggedIn" class="sr-only-sm-and-down">
```
`v-if` und dazugehöriges computed `loggedIn` entfernen

carlobeltrame commented on frontend/src/layouts/DefaultLayout.vue:
```diff
@@ -1,64 +1,54 @@
+            <v-icon left>mdi-format-list-bulleted-triangle</v-icon>
+            <span>Meine Camps</span>
+          </v-list-item>
+          <v-list-item v-if="loggedIn" block @click="logout">
```
`v-if` entfernen

carlobeltrame commented on frontend/src/layouts/DefaultLayout.vue:
```diff
@@ -69,7 +59,7 @@
     </v-content>
 
-    <v-bottom-navigation v-if="$vuetify.breakpoint.xs" fixed grow>
+    <v-bottom-navigation v-if="loggedIn && $vuetify.breakpoint.xs" fixed grow>
```
`loggedIn` entfernen

carlobeltrame reviewed (changes requested):
Puh, grosse Änderung, viele Kommentare. Sieht echt sehr cool aus muss ich sagen!
Falls du keine Zeit für die Änderungen hast meldest du dich, dann mach ich dir einen PR in deinen Branch so wie ich es mir vorstelle.

manuelmeister committed:
Implemented the filter as instructed by @carlobeltrame

https://github.com/ecamp/ecamp3/pull/376#discussion_r387620285
https://github.com/ecamp/ecamp3/pull/376#discussion_r387718631

usu commented on frontend/src/components/camp/Basic.vue:
```diff
@@ -3,24 +3,15 @@ Displays details on a single camp and allows to edit them.
     <v-card-text v-else>
       <v-form>
-        <v-text-field
+        <api-input
```
`[Vue warn]: Missing required prop: "uri"`

Siehst du, Vue ist gleicher Meinung wie ich ;-)
Nehmen wir das in die nächste Sitzung?

manuelmeister replied:
Jap

usu commented on backend/module/eCampApi/src/eCampApi/V1/Rpc/Index/IndexController.php:
```diff
@@ -37,6 +37,14 @@ public function indexAction() {
+                'route' => [
+                    'name' => 'e-camp-api.rest.doctrine.user',
+                    'params' => [ 'user_id' => $userId ]
+                ]
```
Du verlinkst hier auf die user-Ressource. Dann bekommst du aber nur die extern sichtbaren User-Daten und nicht die für den User selbst ersichtlichen Daten. Ist das so gewollt? Die E-Mailadresse ist dort zwar noch drin, aber konsequenterweise müsse man sich überlegen, ob wir die nicht entfernen wollen. Siehe auch Diskussion unter https://github.com/ecamp/ecamp3/issues/342

manuelmeister replied:
@usu Wie müsste das den korrekt verlinkt werden?

usu replied:
Lass uns für die nächste Sitzung aufsparen. Die Frage ist, was die Erwartung ist, welche Daten unter Profile verlinkt sind. Ich hätte jetzt eher tendiert, die bisherige Struktur unter dem `auth` Endpoint zu nutzen. Aber besser, wenn wir das kurz gemeinsam durchsprechen.

manuelmeister replied:
Okay

usu commented on frontend/src/components/Picasso.vue:
```diff
@@ -3,28 +3,30 @@ Listing all given event instances in a calendar view.
+    :weekday-format="weekdayFormat"
+    :weekdays="[1, 2, 3, 4, 5, 6, 0]"
+    color="primary"
+    @click:event="showEventInstance" />
```
Was genau macht `:event`? Hab ich so noch nicht gesehen.

Ausserdem erhalte ich in der Konsole folgende Fehlermeldung:
`[Vue warn]: The .native modifier for v-on is only valid on components but it was used on <div>.`
Kriegst du die auch?

carlobeltrame replied:
Ich glaube `click:event` könnte man auf Deutsch etwa mit `klick:Block` übersetzen. Es ist der Vue event der emitted wird wenn man auf einen Kalender-Event (`:event`) klickt (`click`)

carlobeltrame replied:
Der Konsolen-Fehler ist ein Problem mit vuetify-Kalendern und Vue 2.6.11, siehe https://github.com/vuetifyjs/vuetify/issues/9999 und https://github.com/vuejs/vue/pull/11052

usu replied:
Ah , now I see. Doppelpunkt ist erlaubt im event-Namen und `event` meint Kalender-Event und nicht Vue-Event. Danke für die Klärung. Und merci für den Link auf den Vue-Bug, good to know.

usu commented on frontend/src/components/camp/CampPicasso.vue:
```diff
@@ -57,3 +106,21 @@ export default {
 }
 </script>
+
+<style lang="scss">
```
Scoped nicht möglich? Sonst könnts dann mal naming conflicts geben.

usu commented on frontend/src/components/camp/Basic.vue:
```diff
@@ -3,24 +3,15 @@ Displays details on a single camp and allows to edit them.
-      </v-toolbar-title>
-      <v-spacer />
-    </v-toolbar>
+  <card-view title="Admin">
```
Mal als Vorwarnung: Das wird den ein oder anderen Merge-conflict geben, je nach dem wer schneller ist :-)
Beim Pull request von Forte und mir gibts Basic.vue nicht mehr:
https://github.com/ecamp/ecamp3/pull/382/files

manuelmeister replied:
Von mir aus kann ich auch nach euren Pull Requests mergen. Dann bringe ich das in die gewünschte Form.

usu commented on frontend/src/components/form/ApiInput.vue:
```diff
@@ -4,64 +4,56 @@ You can two-way bind to the value using v-model.
-      </v-form>
-    </span>
-  </span>
+  <div class="mb-4">
```
Konflikt auch hier: `ApiInput` gibts nach dem Pull Request https://github.com/ecamp/ecamp3/pull/372/files nicht mehr.

usu commented on frontend/src/layouts/DefaultLayout.vue:
```diff
@@ -1,64 +1,54 @@
-        <v-icon>mdi-account</v-icon>
-      </v-btn>
+      <global-search />
+      <v-menu offset-y dark
```
Da gibt es viel Redundanz zu `CampLayout`. Wenn wir mit 2 Layouts gehen, dann wär cool wenn wir gleichen Code in Komponenten auslagern so dass wir möglichst keinen kopierten Code haben.

usu commented on frontend/src/views/DayPicasso.vue:
```diff
@@ -10,7 +10,7 @@
 
 <script>
-import Picasso from '@/components/Picasso'
+import Picasso from '../components/Picasso'
```
Ich würde generell eher mit `@` als mit `..` arbeiten. Dann bist du unabhängig davon, wo sich `DayPicasso` genau befindet --> weniger refactoring beim Verschieben von ganzen Vue-Komponenten

manuelmeister replied:
Okay, bei mir hat PhpStorm das @ nicht aufgelöst. Konnte das Problem aber jetzt mit https://stackoverflow.com/questions/51588009/vue-js-configuring-webstorm-to-set-in-path-files-to-the-src-folder
lösen.

usu reviewed (changes requested):
Finde sieht cool aus. Viele Änderungen hinein gepackt --> ich denke es wär gut, wenn du die Hauptkomponenten der anderen grossen Pull requests möglichst wenig berührst. Sonst haben wir dann den Merge-Salat.

manuelmeister committed:
Changed structure to only use the `vue-router` and no layouts, which improves rendering and loading between states.

manuelmeister committed:
Renamed CardView.vue to ContentCard.vue and moved Picasso.vue and CampPicasso.vue to right place

manuelmeister committed:
isLoggedIn entfernt, da wir hier bereits eingeloggt sind.

manuelmeister committed:
Nur eine Periode zeigen. Evtl muss die Periode in Route aufgenommen werden.

manuelmeister committed:
Rearranged

manuelmeister committed:
Added character variants so that the characters are more easily distinguishable

manuelmeister committed:
Removed unnecessary definitions

manuelmeister committed:
Removed unnecessary loggedin

manuelmeister committed:
Extracted MobileSearch.vue from WeekPicasso.vue and renamed Search.vue to DesktopSearch.vue

manuelmeister committed:
Now with correct `v-app` usage, the sizing is correct

manuelmeister committed:
Sidebar consistency improved

manuelmeister committed:
Now with correct `v-app` usage, the sizing is correct

manuelmeister committed:
Removed unnecessary definition

manuelmeister committed:
Improved naming of view components

manuelmeister committed:
Store format in query. Also added parse Boolean query parameter function

manuelmeister committed:
Made route naming consistent

manuelmeister committed:
Removed username from auth.js, exchanged it for profile and removed references to route picasso

manuelmeister commented:
Ich habe mal versucht die Layout zu entfernen und jetzt navigiert es sich viel flüssiger.
@usu @carlobeltrame 
Schaut es euch mal an.

manuelmeister committed:
Merge branch 'devel' of github.com:ecamp/ecamp3 into consistent-navigation

manuelmeister committed:
Use correct api input

manuelmeister committed:
Renamed CampHome to Admin

manuelmeister committed:
Fixed campRoute without subroute

manuelmeister committed:
Fixed problems and improved consistency with merged features from devel

manuelmeister committed:
Fixed layout of the forms

manuelmeister committed:
Improved spacing

manuelmeister committed:
Optimized collaborator view

manuelmeister committed:
Optimized loading states

manuelmeister committed:
Removed class

manuelmeister committed:
Improved buttons. Added possibility for a custom icon

manuelmeister committed:
Improved collaborators

manuelmeister committed:
Fixed spacing problem in form

manuelmeister committed:
Moved Collaborators.vue into view

manuelmeister committed:
Moved SideBar.vue into components

manuelmeister committed:
Moved Collaborators.vue into view

manuelmeister committed:
Moved SideBar.vue into components

manuelmeister committed:
Merge branch 'devel' of github.com:ecamp/ecamp3 into consistent-navigation

# Conflicts:
#	frontend/src/store/storeValueProxy.js

manuelmeister committed:
Fixed snapshots

manuelmeister committed:
Fixed snapshots

manuelmeister committed:
Fixed test, is wrong assumption

manuelmeister commented:
@carlobeltrame @usu könnt ihr den bitte nochmals anschauen? Brauche dann wohl noch eine Einführung ins Testing.

carlobeltrame commented on frontend/src/components/base/ContentGroup.vue:
```diff
@@ -0,0 +1,34 @@
+<!--
+Displays the content wrapped inside a card.
```
Bitte anpassen

carlobeltrame commented on frontend/src/components/base/MobileSearch.vue:
```diff
@@ -0,0 +1,53 @@
+<template>
+  <v-sheet v-if="$vuetify.breakpoint.xs" min-height="56">
```
Mir würde es besser gefallen wenn das `v-if` hier entfernst und die Verantwortung "anzeigen oder nicht" an die Komponente überträgst die die MobileSearch anzeigt (also CampOverview)

carlobeltrame commented on frontend/src/components/base/MobileSearch.vue:
```diff
@@ -0,0 +1,53 @@
+<template>
+  <v-sheet v-if="$vuetify.breakpoint.xs" min-height="56">
+    <v-app-bar
+      v-if="$vuetify.breakpoint.xs"
```
Dieses `v-if` ist momentan gänzlich nutzlos, aber wenn du das äussere `v-if` von hier entfernst könnte es theoretisch einen Unterschied machen. Ist es in der Praxis nützlich, hier zu unterscheiden?

carlobeltrame commented on frontend/src/components/base/MobileSearch.vue:
```diff
@@ -0,0 +1,53 @@
+      overlay-color="white" hide-overlay
+      fullscreen transition="dialog-top-transition">
+      <v-card>
+        <v-sheet v-if="$vuetify.breakpoint.xs" height="70" />
```
Same here

carlobeltrame commented on frontend/src/components/base/SideBar.vue:
```diff
@@ -0,0 +1,40 @@
+  data () {
+    return {
+      drawer: false,
+      mini: !this.$vuetify.breakpoint.mdAndUp
```
Sollte das nicht eher ein computed sein? Wenn es data ist dann reagiert es nicht auf eine Änderung der Fenstergrösse (oder?)

manuelmeister replied:
Ja die Frage ist, ob man will, dass es einfach zuklappt, wenn man das Fenster verkleinert.

carlobeltrame commented on frontend/src/components/camp/EventList.vue:
```diff
@@ -0,0 +1,44 @@
+<!--
+Listing all given event instances in a calendar view.
```
Bitte aktualisieren

carlobeltrame commented on frontend/src/components/base/Logo.vue:
```diff
@@ -0,0 +1,32 @@
+        <i>🏕</i>️<span v-if="text" class="ml-4 mr-2">eCamp</span>
+      </v-toolbar-title>
+    </v-btn>
+    <slot/>
```
```suggestion
    <slot />
```
Könntest du noch `npm run lint --fix` bzw. `docker-compose exec frontend npm run lint --fix` drüber laufen lassen? Hat einige Linter-Fehler bei mir.

manuelmeister replied:
Habe ich gemacht. Ist aber beides akzeptiert.

carlobeltrame replied:
Hm wie kann das sein? Dann stimmt etwas mit unserer (meiner?) ESLint Konfiguration nicht

carlobeltrame commented on frontend/src/views/TopBar.vue:
```diff
@@ -0,0 +1,88 @@
+        <v-list-item :to="{ name: 'login' }" block>
+          <v-icon left>mdi-login</v-icon>
+          <span>Log in</span>
+        </v-list-item>
```
Das sollte stattdessen ein Logout-Link sein

carlobeltrame commented on frontend/src/router.js:
```diff
@@ -12,86 +12,107 @@ export default new Router({
+          beforeEnter: parseBooleanInQuery
         },
         {
-          path: '',
```
Du hast die Route mit dem leeren Pfad entfernt. Was jetzt also nicht mehr geht ist /camps/eafb123/So-La-2020. Finde ich vertretbar, auch wenn ich lieber /camps/eafb123/So-La-2020 als /camps/eafb123/So-La-2020/overview hätte.

manuelmeister replied:
Ich habe nicht herausgefunden, wie ich diesen Pfad umleiten kann, so dass aber nicht bei einem Link mit Query (?list=false) noch zusätzlich ein /camps/eafb123/So-La-2020/overview/**overview** hinzugefügt.

carlobeltrame replied:
```js

      children: [
        {
          path: 'collaborators',
          name: 'camp/collaborators',
          component: () => import(/* webpackChunkName: "campCollaborators" */ './views/camp/Collaborators.vue')
        },
        {
          path: 'admin',
          name: 'camp/admin',
          component: () => import(/* webpackChunkName: "campAdmin" */ './views/camp/Admin')
        },
        {
          path: '',
          name: 'camp/overview',
          alias: 'overview',
          component: () => import(/* webpackChunkName: "campPicasso" */ './views/camp/CampOverview'),
          beforeEnter: parseBooleanInQuery
        }
      ]
```
So geht alles was ich gerade teste, übersehe ich etwas? Den Alias kannst du theoretisch auch weglassen, weil der eh nie in den verlinkten URLs stehen wird (ausser man schreibts von Hand dazu).
Wichtig ist aber dass der Eintrag mit leerem `path` am Ende der Liste steht. Nur so wird /camps/4636d1c8/admin korrekt verarbeitet.

carlobeltrame commented on frontend/src/router.js:
```diff
@@ -12,86 +12,107 @@ export default new Router({
+          path: 'overview',
+          name: 'camp/overview',
+          component: () => import(/* webpackChunkName: "campPicasso" */ './views/camp/CampOverview'),
+          beforeEnter: parseBooleanInQuery
```
Ist das wirklich nötig? Könntest du nicht in CampOverview.vue stattdessen `this.$route.query.list === 'true'` überprüfen?

manuelmeister replied:
Das hat leider nicht funktioniert. Weil
```javascript 
true
```
≠
```javascript
"true"
```

Und ich setzte es ja als Boolean in den Query

carlobeltrame replied:
Okay, I see. Den False state könnten wir aber durch Abwesenheit des Query Params anzeigen. So gehts ohne parseBooleanInQuery:
```
:to="{ query: { ...this.$route.query, list: !this.listFormat || undefined } }">
```
Das `|| undefined` bewirkt, dass der Query Parameter ganz entfernt wird wenn er false sein sollte. Dann kommt man auch mit dem Check `this.$route.query.list` aus (ohne `=== 'true'`).

manuelmeister replied:
`list: !this.listFormat` ist dann eigentlich ein `!(this.listFormat == 'true')` und nicht `===` oder?

manuelmeister replied:
Wenn man jetzt in der URL ?list=false eingibt wird trotzdem eine Liste angezeigt. Ist das uns egal?

carlobeltrame replied:
Wenn du die `''` um `true` aus deiner Aussage rausnimmst stimme ich dir zu.

carlobeltrame replied:
Einziger Nachteil: Wenn der User von Liste auf Wochenansicht umschaltet bleibt im `query` objekt der Key `list: undefined`. Das ist an sich kein Problem, ausser beim `exact` route matching welches wir fürs Highlighting des `v-btn` in der TopBar verwenden. Konkret: Wenn du auf den Button drückst ist der v-btn in der TopBar nicht mehr highlighted. Aber das ist bereits jetzt so wenn man in der Listenansicht ist (weil der Query Param nicht exact matched).

Als Lösung würde ich entweder das Highlighting für diesen Button ganz ausschalten (wie auch immer, Vuetify macht das automatisch rein bei `:to`), oder von Hand die active class hinzufügen: `:class="{ 'v-btn--active': someComputedConditionThatIgnoresTheQueryParams }"`
Siehe auch: https://forum.vuejs.org/t/how-to-remove-one-query-string-from-url/39176/4 und https://github.com/vuejs/vue-router/issues/2040

manuelmeister replied:
Habe das Problem gelöst, indem ich einen Redirect von `path: ''` auf `camp/overview` mache und beim Button `exact` entfernt habe. So ist `/overview` standardmässig drin. Finde ich verständlicher, als mit dem `alias`.

carlobeltrame replied:
> Wenn man jetzt in der URL ?list=false eingibt wird trotzdem eine Liste angezeigt. Ist das uns egal?

Ja das wäre mir egal, denn die URLs werden ja eigentlich immer von Vue korrekt generiert. Ich finde wir müssen es nicht unterstützen dass der User list=false tippen kann wenn er den list-Parameter genausogut weglöschen kann.

carlobeltrame replied:
> Habe das Problem gelöst, indem ich einen Redirect von `path: ''` auf `camp/overview` mache und beim Button `exact` entfernt habe. So ist `/overview` standardmässig drin. Finde ich verständlicher, als mit dem `alias`.

Ich sehe im Code aber aktuell kein `path: ''` und auch keinen Redirect. Wie ist es denn jetzt wirklich?

Und wenn du `exact` entfernst ist der Button eben auch aktiv wenn man in /admin oder /collaborators ist, ist dir das bewusst?

carlobeltrame commented on frontend/src/components/camp/CollaboratorListItem.vue:
```diff
@@ -0,0 +1,40 @@
+      <v-list-item-title>{{ collaborator.user().username }}</v-list-item-title>
+      <v-list-item-subtitle>{{ collaborator.user().mail }}</v-list-item-subtitle>
+    </v-list-item-content>
+    <api-select
```
Ich sehe hier kein kleines Dropdown-Pfeilchen, ist das so beabsichtigt? (Vielleicht war das auch vorher schon so)

manuelmeister replied:
Das war vorher auch schon so. Würde ich in ein weiteres Issue aufnehmen.

carlobeltrame commented on frontend/src/views/camp/CampOverview.vue:
```diff
@@ -0,0 +1,99 @@
+<!--
+Listing all event instances of a single period.
```
Update this

manuelmeister committed:
Move v-if of MobileSearch.vue to CampOverview.vue

manuelmeister committed:
Fixed description

manuelmeister committed:
Linted whole project

manuelmeister committed:
Changed button to logout

usu commented on frontend/src/components/base/BackButton.vue:
```diff
@@ -0,0 +1,16 @@
+<template>
+  <v-btn v-if="$vuetify.breakpoint.xsOnly" aria-label="Zurück"
```
Wieso wird der Button nur in `xs` Display angezeigt?

manuelmeister replied:
Auf Desktop hat er mich gestört. War aber nur ein Bauchgefühl. (Kann ich gerne ändern)

usu commented on frontend/src/components/base/ButtonAdd.vue:
```diff
@@ -12,7 +13,10 @@
-  name: 'ButtonAdd'
+  name: 'ButtonAdd',
+  props: {
+    icon: { type: String, default: 'mdi-plus' }
```
👍

usu commented on frontend/src/components/form/ApiWrapper.vue:
```diff
@@ -5,7 +5,7 @@ Wrapper component for form components to save data back to API
 <template>
   <v-form
     inline
-    class="mb-2">
```
Wollen wir nicht einen Standardabstand gesetzt lassen, den man dann halt überschreiben kann? Jetzt hast du fast bei jedem Aufruf `mb-4` oder `my-4` mit drin.

manuelmeister replied:
Das Problem ist, wenn du das ganze zb in CollaboratorListItem in der Liste anzeigen möchtest ist das Margin störend

usu replied:
Ok. Aber wär es nicht einfacher in diesem spezifischen Einzelfall den Margin zu entfernen und ansonsten einen Standardmargin drin zu haben?

usu commented on frontend/src/components/form/ApiWrapper.vue:
```diff
@@ -16,19 +16,21 @@ Wrapper component for form components to save data back to API
+      small elevation="0"
+      color="warning" tile
+      class="mb-0"
+      height="auto"
```
Ich versteh die Idee dahinter, aber die Buttons sind jetzt schon mega gross (insbesondere beim Textarea)...

manuelmeister replied:
Ja aber das ist ja nur wenn es nicht automatisch gespeichert wird. (Was wohl fast nie der Fall ist. Da wir ja eher autosave und Formular haben)

usu replied:
In den meisten Fällen ja. Eventuell wird es Ausnahmen geben (z.B. ein Kommentarfeld)? Dann würde ich mir schon eher Buttons wünschen wie z.B. hier bei Github.

manuelmeister replied:
Okay, dann schaue ich dass für die ApiTextArea der Stil anders ist: #393

manuelmeister committed:
Fixed eslint problem

usu commented on frontend/src/views/BottomBar.vue:
```diff
@@ -0,0 +1,19 @@
+
+<script>
+export default {
+  name: 'BottomBar'
```
Vom Naming her würde ich eher etwas genauer beschreiben, z.B. `BottomBarMobile`, da der Inhalt dieser Komponente ja nur für `xs` relevant ist. Können wir von mir aus aber auch als Sitzungsthema nehmen "Orderstruktur & Komponentennaming". Da haben wir im Moment noch etwas Unordnung drin.

manuelmeister replied:
Ja finde ich auch noch nicht ganz eindeutig. Nehme ich auf: #389

usu commented on frontend/src/views/Profile.vue:
```diff
@@ -0,0 +1,40 @@
+export default {
+  name: 'Home',
+  components: {
+    ApiTextField: () => import('@/components/form/ApiTextField'),
```
Würde ich eher nicht dynamisch laden. Das führt zu unnötig vielen Netzwerkrequests. Ich glaubs wir sollten uns für das dynamische Laden auf den Router beschränken. Oder ist die Meinung, dass man das möglichst aufsplittet und dann von Caching profitiert?

manuelmeister replied:
Ich weiss es nicht. Schauen wir wohl am Besten in #377 an

usu commented on frontend/src/views/TopBar.vue:
```diff
@@ -0,0 +1,88 @@
+import DesktopSearch from '@/components/base/DesktopSearch'
+
+export default {
+  name: 'TopBar',
```
TopBarDesktop?

usu commented on frontend/src/views/camp/Periods.vue:
```diff
@@ -0,0 +1,49 @@
+import SideBar from '@/components/base/SideBar'
+
+export default {
+  name: 'CampPeriods',
```
Da der Inhalt fix eine Sidebar anzeigt, würde ich das im Namen reflektieren, z.B. `SideBarPeriods`.

usu commented on frontend/src/views/camp/TopBar.vue:
```diff
@@ -0,0 +1,105 @@
+    </v-toolbar-items>
+    <v-spacer />
+    <desktop-search />
+    <v-menu offset-y dark
```
Keep it DRY. Das dürfte fast 1:1 der gleiche Code sein wie in der anderen `TopBar`. Eigene Komponente `TopBarUserMenu`?

usu reviewed (commented):
Ich erhalte diverse `eslint` warning. Könntest du noch den linter laufen lassen? Oder deinen Editor konfigurieren, dass er gleich autolint macht beim Speichern?

manuelmeister committed:
Extracted UserMenu

manuelmeister committed:
Specified naming of components

manuelmeister committed:
Removed condition on BackButton.vue

manuelmeister committed:
Adjusted name of the components

manuelmeister committed:
Changed mini setting to computed

manuelmeister committed:
Implemented suggestions from @carlobeltrame

Signed-off-by: Manuel Meister <news.manuelsworld@gmail.com>

manuelmeister committed:
Solved active button problem with query params

manuelmeister commented on frontend/src/router.js:
```diff
@@ -90,16 +90,19 @@ export default new Router({
-          name: 'camp/admin',
-          component: () => import(/* webpackChunkName: "campAdmin" */ './views/camp/Admin')
+          path: '',
+          redirect: { name: 'camp/overview' }
```
@carlobeltrame hier ist der redirect

manuelmeister committed:
The router-definition can't really be tested

carlobeltrame commented on frontend/src/views/NavDesktop.vue:
```diff
@@ -0,0 +1,59 @@
+<script>
+import { campFromRoute, campRoute } from '@/router'
+import DesktopSearch from '@/components/base/NavDesktopSearch'
+import TopBarUserMenu from '@/components/base/NavDesktopUserMenu'
```
```suggestion
import NavDesktopUserMenu from '@/components/base/NavDesktopUserMenu'
```

carlobeltrame commented on frontend/src/views/NavDesktop.vue:
```diff
@@ -0,0 +1,59 @@
+export default {
+  name: 'NavDesktop',
+  components: {
+    TopBarUserMenu,
```
```suggestion
    NavDesktopUserMenu,
```

carlobeltrame commented on frontend/src/views/camp/NavDesktop.vue:
```diff
@@ -0,0 +1,59 @@
+<script>
+import { campFromRoute, campRoute } from '@/router'
+import DesktopSearch from '@/components/base/NavDesktopSearch'
+import TopBarUserMenu from '@/components/base/NavDesktopUserMenu'
```
```suggestion
import NavDesktopUserMenu from '@/components/base/NavDesktopUserMenu'
```

carlobeltrame commented on frontend/src/views/camp/NavDesktop.vue:
```diff
@@ -0,0 +1,59 @@
+export default {
+  name: 'NavDesktop',
+  components: {
+    TopBarUserMenu,
```
```suggestion
    NavDesktopUserMenu,
```

carlobeltrame commented on frontend/src/components/base/NavDesktopUserMenu.vue:
```diff
@@ -0,0 +1,64 @@
+      logoutIcon: 'mdi-logout'
+    }
+  },
+  methods: {
```
```suggestion
  props: {
    username: { type: String, required: true }
  },
  methods: {
```

manuelmeister replied:
Added as a computed. Doesn't need to be a prop, because it isn't used in NavDesktop.

carlobeltrame commented on frontend/src/views/event/DayOverview.vue:
```diff
@@ -0,0 +1,55 @@
+               :camp="period.camp"
+               :event-instances="eventInstances.items"
+               :start="startOfDay"
+               interval-height="36"
```
```suggestion
               :interval-height="36"
```
Otherwise, "36" is passed in as string (should be a number).

carlobeltrame reviewed (changes requested):
Sieht sehr gut aus jetzt, mit dem Router funktioniert alles wunderbar. Good job!
Ich habe noch zwei Konsolen-Fehler gefunden und dir die Fixes dafür hier vorgeschlagen. Wenn du das gemacht hast ist das auch von mir aus okay :+1:

manuelmeister committed:
Implemented suggestions from @usu
https://github.com/ecamp/ecamp3/pull/376#discussion_r394118460

manuelmeister committed:
Fixed console error
Loading behaviour should be improved #390

manuelmeister committed:
Fixed naming

manuelmeister committed:
Removed unnecessary styles

manuelmeister committed:
Used ContentCard in aside

manuelmeister committed:
Fixed wrong type

manuelmeister committed:
Improved consistency in naming

manuelmeister committed:
Renamed overview to program

manuelmeister committed:
Removed last reference to CardView
(was replaced by ContentView)

manuelmeister committed:
Added test for vue loading filter
